diff --git a/db.sqlite3 b/db.sqlite3
index 728e9ed..869981e 100644
Binary files a/db.sqlite3 and b/db.sqlite3 differ
diff --git a/ikaya_oscar_apps/catalogue/views.py b/ikaya_oscar_apps/catalogue/views.py
index 561fd9d..9a72d67 100644
--- a/ikaya_oscar_apps/catalogue/views.py
+++ b/ikaya_oscar_apps/catalogue/views.py
@@ -17,11 +17,24 @@ from django import template
 def currency_conv_view(request):
     currency_code = request.GET['currency_code']
     factor = {}
-    factor = {'INR':1.00, 'USD':0.015, 'GBP':0.011, 'EUR':0.014}
+    try: 
+        req = urllib2.urlopen('http://finance.yahoo.com/d/quotes.csv?e=.csv&f=sl1d1t1&s=INR'+currency_code+'=X') 
+        result = req.read() 
+        factor.update({currency_out:result.split(',')[1]}) 
+    except:
+        factor = {'INR':1.00, 'USD':0.0155, 'GBP':0.0115, 'EUR':0.0134}
+
     request.session['currency_code'] = currency_code 
     request.session['currency_factor'] = factor[currency_code]
     return HttpResponseRedirect(request.META['HTTP_REFERER'])
 
+#    currency_code = request.GET['currency_code']
+#    factor = {}
+#    factor = {'INR':1.00, 'USD':0.014, 'GBP':0.014, 'EUR':0.013}
+#    request.session['currency_code'] = currency_code 
+#    request.session['currency_factor'] = factor[currency_code]
+#    return HttpResponseRedirect(request.META['HTTP_REFERER'])
+
 class CatalogueView(TemplateView):
     """
     Browse all products in the catalogue
@@ -120,4 +133,4 @@ class ProductCategoryView(TemplateCategoryView):
         else:
             pass
 
-        return ctx
\ No newline at end of file
+        return ctx
diff --git a/templates/oscar/checkout/shipping_address.html b/templates/oscar/checkout/shipping_address.html
index efffbb7..5eddfdd 100644
--- a/templates/oscar/checkout/shipping_address.html
+++ b/templates/oscar/checkout/shipping_address.html
@@ -56,7 +56,7 @@
                                         {% if address.is_default_for_shipping %}
                                             <button type="submit" class="btn btn-success btn-large ship-address" data-loading-text="{% trans 'Saving...' %}"><i class="icon-ok"></i> {% trans "Ship to your default shipping address" %}</button>
                                         {% else %}
-                                            <button type="submit" class="btn btn-primary btn-large ship-address" data-loading-text="{% trans 'Saving...' %}">{% trans "Ship to this address" %}</button>
+                                            <button type="submit" class="btn save pull-left btn-primary btn-large ship-address" data-loading-text="{% trans 'Saving...' %}">{% trans "Ship to this address" %}</button>
                                         {% endif %}
 
                                         <div class="btn-group address-controls">
